bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/MetadataDrivers.java

```diff
@@ -369,7 +369,8 @@ public static MetadataBookieDriver getBookieDriver(URI uri) {
     public static <T> T runFunctionWithRegistrationManager(ServerConfiguration conf,
                                                            Function<RegistrationManager, T> function)
             throws MetadataException, ExecutionException {
-        return runFunctionWithMetadataBookieDriver(conf, driver -> function.apply(driver.getRegistrationManager()));
+        return runFunctionWithMetadataBookieDriver(
+                conf, driver -> function.apply(driver.createRegistrationManager()));
```

Comment:
will the instance returned by  createRegistrationManager be closed ?

Reply:
Fixed.